src/System.Net.Security/src/System/Net/Unix/SslConnectionInfo.cs

```diff
@@ -15,30 +15,31 @@ internal class SslConnectionInfo
         public readonly int KeyExchangeAlg;
         public readonly int KeyExchKeySize = 0;
 
-        internal SslConnectionInfo(Interop.libssl.SSL_CIPHER cipher)
+        internal SslConnectionInfo(Interop.libssl.SSL_CIPHER cipher, string protocol)
         {
-            Protocol = (int) MapProtocol(cipher.algorithm_ssl);
+            Protocol = (int) MapProtocolVersion(protocol);
             DataCipherAlg = (int) MapCipherAlgorithmType(cipher.algorithm_enc);
             KeyExchangeAlg = (int) MapExchangeAlgorithmType(cipher.algorithm_mkey);
             DataHashAlg = (int) MapHashAlgorithmType(cipher.algorithm_mac);
             // TODO (Issue #3362) map key sizes
         }
 
-        private static SslProtocols MapProtocol(long algorithm)
+        private SslProtocols MapProtocolVersion(string protocolVersion)
         {
-            switch (algorithm)
+            switch (protocolVersion)
             {
-                case 1:
+                case "SSLv2":
```

Comment:
Yes. libssl needs to be explicitly rebuilt from sources enabling it. But I don't see what this comment has to do with the change referenced.